### PR TITLE
Updating redirect code

### PR DESCRIPTION
**Pull request by fangyu3**

### CODE REVIEW

**Comment by fangyu3** [comment]
@bugatti12345 
Now the redirect works, but it gives this error whenever i sign up/login the first time after restarting the client & server. If i try it afterwards, it's working as intended. I am not sure what the issue is here.  Can you take a look at the code and see if im doing things right?

Screen Shot 2021-03-31 at 4 52 21 PM

File: client/src/components/Header.js [hunk 1]
```diff
@@ -79,6 +81,10 @@ function Header() {
             type:"success"
             })   
           setAnchorEl(null)
+          userContext.setIsSubscribed(false)
```
Comment (fangyu3) [inline client/src/components/Header.js:84]: reset user context when logging out

Comment (JoshL579) [reply client/src/components/Header.js:84]: > @bugatti12345
> Now the redirect works, but it gives this error whenever i sign up/login the first time after restarting the client & server. If i try it afterwards, it's working as intended. I am not sure what the issue is here. Can you take a look at the code and see if im doing things right?
> 
> Screen Shot 2021-03-31 at 4 52 21 PM

@fangyu3 I don't think you did anything wrong, but can you maybe try change the `redirectOnLogin()` in line 131 into `redirectOnLogin`?

Comment (JoshL579) [reply client/src/components/Header.js:84]: However the error is not related to redirect part tho...

File: client/src/pages/Login.js [hunk 2]
```diff
@@ -95,16 +88,11 @@ function Login() {
...
                     user.setUserEmail(res.userEmail)
-                    
-                    if (res.user_url.length === 0 || res.user_timezone.length === 0)
-                        history.push("/onboarding/profile-settings");
-                    else if (res.user_available_day.length === 0 || res.user_available_time.length === 0)
-                        history.push("/onboarding/availability")
-                    else
-                        history.push("/home");
+                    user.setOnboardingStep(res.onboardingStep)
+                    user.setUserId(res.id);
```
Comment (JoshL579) [inline client/src/pages/Login.js:95]: @fangyu3 can you maybe nest these into a return?
like:
```javascript
return(
  user.setIsSubscribed(res.isSubscribed);
  user.setUserEmail(res.userEmail);
  user.setOnboardingStep(res.onboardingStep);
  user.setUserId(res.id);
)
```

Comment (JoshL579) [reply client/src/pages/Login.js:95]: don't know if need `{}` here

**Comment by JoshL579** [comment]
Sovled

**Pull request closed without merging**